### PR TITLE
flash_map; add flash_area_to_subareas().

### PR DESCRIPTION
**Pull request by mkiiskila**
Simpler solution when memory use of FCB is of concern. Creates a flash_area array suitable for passing to FCB out of flash_map index, caller controls the number of array entries created.

### CODE REVIEW

File: sys/flash_map/include/flash_map/flash_map.h [hunk 1]
```diff
@@ -106,6 +106,13 @@ uint32_t flash_area_erased_val(const struct flash_area *fa);
  */
 int flash_area_to_sectors(int idx, int *cnt, struct flash_area *ret);
 
+/*
+ * Given flash map index, return info about sectors within the area.
+ * If the physical sector count within region exceeds that, collect
```
Comment (kasjer) [inline sys/flash_map/include/flash_map/flash_map.h]: Second sentence is not that clear,

File: sys/flash_map/src/flash_map.c [hunk 2]
```diff
@@ -131,6 +131,73 @@ flash_area_getnext_sector(int id, int *sec_id, struct flash_area *ret)
...
+        }
+    }
+    if (*cnt > max_cnt) {
+        min_size = fa->fa_size / max_cnt;
+    } else {
+        min_size = 0;
+    }
+
+    next = 1;
+    for (i = first_idx, j = 0; i < last_idx + 1; i++) {
```
Comment (kasjer) [inline sys/flash_map/src/flash_map.c]: could be i <= last_idx

File: sys/flash_map/src/flash_map.c [hunk 3]
```diff
@@ -131,6 +131,73 @@ flash_area_getnext_sector(int id, int *sec_id, struct flash_area *ret)
...
+
+    hf = hal_bsp_flash_dev(fa->fa_device_id);
+    for (i = 0; i < hf->hf_sector_cnt; i++) {
+        hf->hf_itf->hff_sector_info(hf, i, &start, &size);
+        if (start >= fa->fa_off && start < fa->fa_off + fa->fa_size) {
+            if (first_idx == -1) {
+                first_idx = i;
+            }
+            last_idx = i;
+            *cnt = *cnt + 1;
```
Comment (kasjer) [inline sys/flash_map/src/flash_map.c]: looks like *cnt is assigned in line 196, so line 165 is not really needed

Comment (mkiiskila) [reply sys/flash_map/src/flash_map.c]: On close inspection, realized that it is used as an intermediate variable, and used in line 168.

Comment (kasjer) [reply sys/flash_map/src/flash_map.c]: Maybe adding local variable for this would make it simpler and probably would reduce code size as well.

**Comment by ccollins476ad** [comment]
These changes look to be obsolete.  Please reopen as needed.

**Pull request closed without merging**